src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs

```diff
@@ -0,0 +1,362 @@
+                DisableOtherGeometryButtons(MultipointButton);
+
+                // Disable the combo box as this is always a vertex tool when creating a point.
+                ToolComboBox.IsEnabled = false;
+
+                _geometryEditor.Start(GeometryType.Multipoint);
+            }
+        }
+
+        // Start the geometry editor with the polyline geometry type.
+        private void PolylineButton_Click(object sender, RoutedEventArgs e)
+        {
+            if (!_geometryEditor.IsStarted)
+            {
+                // Disable buttons to reflect that the geometry editor has started.
+                DisableOtherGeometryButtons(PolylineButton);
+
+                _geometryEditor.Start(GeometryType.Polyline);
+            }
+        }
+
+        // Start the geometry editor with the polygon geometry type.
+        private void PolygonButton_Click(object sender, RoutedEventArgs e)
+        {
+            if (!_geometryEditor.IsStarted)
+            {
+                // Disable buttons to reflect that the geometry editor has started.
+                DisableOtherGeometryButtons(PolygonButton);
+
+                _geometryEditor.Start(GeometryType.Polygon);
+            }
+        }
+
+        // Set the geometry editor tool from the combo box.
+        private void ToolComboBox_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            var keyValuePair = (KeyValuePair<string, GeometryEditorTool>)ToolComboBox.SelectedItem;
+            var tool = (GeometryEditorTool)keyValuePair.Value;
+            _geometryEditor.Tool = tool;
+
+            // Account for case when vertex tool is selected and geometry editor is started with a polyline or polygon geometry type.
+            // Ensure point and multipoint buttons are only enabled when the selected tool is a vertex tool.
+            PointButton.IsEnabled = MultipointButton.IsEnabled = !_geometryEditor.IsStarted && tool is VertexTool;
+        }
+
+        #endregion Event handlers
+
+        #region Edit controls event handlers
+
+        // Undo the last change made to the geometry while editing is active.
+        private void UndoButton_Click(object sender, RoutedEventArgs e)
+        {
+            _geometryEditor.Undo();
+        }
+
+        // Redo the last change made to the geometry while editing is active.
+        private void RedoButton_Click(object sender, RoutedEventArgs e)
+        {
+            _geometryEditor.Redo();
+        }
+
+        // Delete the currently selected element of the geometry editor.
+        private void DeleteSelectedButton_Click(object sender, RoutedEventArgs e)
+        {
+            _geometryEditor.DeleteSelectedElement();
+        }
+
+        // Update an existing graphic or create a new graphic.
+        private void SaveButton_Click(object sender, RoutedEventArgs e)
+        {
+            Geometry geometry = _geometryEditor.Stop();
```

Comment:
Actually Stop() could return a null geometry, maybe add a check? Otherwise, you get null ref below for getting its type to apply an appropriate symbol.